sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -0,0 +1,388 @@
+            return;
+        }
+
+        eventData.getSystemProperties().forEach((key, value) -> {
+            if (Message.RESERVED_SYSTEM_PROPERTIES.contains(key)) {
+                return;
+            }
+
+            final AmqpMessageConstant constant = AmqpMessageConstant.fromString(key);
+
+            if (constant != null) {
+                switch (constant) {
+                    case MESSAGE_ID:
+                        message.setMessageId(value);
+                        break;
+                    case USER_ID:
+                        message.setUserId((byte[]) value);
+                        break;
+                    case TO:
+                        message.setAddress((String) value);
+                        break;
+                    case SUBJECT:
+                        message.setSubject((String) value);
+                        break;
+                    case REPLY_TO:
+                        message.setReplyTo((String) value);
+                        break;
+                    case CORRELATION_ID:
+                        message.setCorrelationId(value);
+                        break;
+                    case CONTENT_TYPE:
+                        message.setContentType((String) value);
+                        break;
+                    case CONTENT_ENCODING:
+                        message.setContentEncoding((String) value);
+                        break;
+                    case ABSOLUTE_EXPIRY_TIME:
+                        message.setExpiryTime((long) value);
+                        break;
+                    case CREATION_TIME:
+                        message.setCreationTime((long) value);
+                        break;
+                    case GROUP_ID:
+                        message.setGroupId((String) value);
+                        break;
+                    case GROUP_SEQUENCE:
+                        message.setGroupSequence((long) value);
+                        break;
+                    case REPLY_TO_GROUP_ID:
+                        message.setReplyToGroupId((String) value);
+                        break;
+                    default:
+                        throw new IllegalArgumentException(
+                            String.format(
+                                "Property is not a recognized reserved property name: %s",
+                                key));
+                }
+            } else {
+                final MessageAnnotations messageAnnotations = (message.getMessageAnnotations() == null)
+                    ? new MessageAnnotations(new HashMap<>())
+                    : message.getMessageAnnotations();
+                messageAnnotations.getValue().put(Symbol.getSymbol(key), value);
+                message.setMessageAnnotations(messageAnnotations);
+            }
+        });
+    }
+
+    private static int getPayloadSize(org.apache.qpid.proton.message.Message msg) {
+
```

Comment:
Should make sure this aligns with the track 1 serialization code for a service bus message. iirc, it's different.